web/js/map-OpenLayers.js

```diff
@@ -1267,16 +1267,17 @@ OpenLayers.Protocol.FixMyStreet = OpenLayers.Class(OpenLayers.Protocol.HTTP, {
             options.params.show_old_reports = 1;
         }
         var page;
+        var qs = fixmystreet.utils.parse_query_string();
         if (this.use_page) {
             page = $('.pagination:first').data('page');
             this.use_page = false;
         } else if (this.initial_page) {
             page = 1;
         } else {
-            var qs = fixmystreet.utils.parse_query_string();
             this.initial_page = page = qs.p || 1;
         }
         options.params.p = page;
+        options.params.zoom = qs.zoom;
```

Comment:
This is getting the zoom parameter from the URL before the zoom takes effect, and so is incorrect (you can see this is you start at zoom 15 and zoom out, you still see defect pins because 15 is sent to the server). I also wonder if there's a case where there might not be a zoom in the URL for some reason. Anyway, I think `fixmystreet.map.getZoom()` returns the current zoom level, no need to look at the URL.